### PR TITLE
Improvement pytest

### PR DESCRIPTION
**Pull request by felddy**

### CODE REVIEW

**Comment by felddy** [comment]
So this PR adds unit tests, and code coverage checks.  I think we should start maturing our test writing chops sooner than later.  As the team grows the tests will give new members some confidence that they aren't committing system breaking changes.  Also, chasing a 100% code-coverage can be a new pastime.  

A bigger change is that I moved `example` into a `src` directory.  
I was convinced of this for a few reasons, one of which was this blog post:
https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure

Let me know what you think.

File: .bandit.yml [hunk 1]
```diff
@@ -10,4 +10,4 @@ tests:
 # - B102
 
 skips:
-# - B101 # skip "assert used" check since assertions are required in pytests
+  - B101 # skip "assert used" check since assertions are required in pytests
```
Comment (jsf9k) [inline .bandit.yml:14]: I was also perturbed by this.  [In another project](https://github.com/cisagov/skeleton-ansible-role/blob/develop/.pre-commit-config.yaml#L56) I opted to make bandit ignore the test directory.

We probably don't want asserts in production code, so I'm very uneasy about disabling the assert check altogether.  I'm more comfortable not running bandit on the tests than doing that, but the point is debatable.

Let's reach a consensus, or find a way to run bandit with the assert check on the library code while running it without the assert check on the test code.

**Review by jsf9k — CHANGES_REQUESTED** [review]
I think these changes are good, but please see my comment.  Ideally we could find a solution that lets us run `bandit` with the assert check enabled on the library code and disabled on the unit test code.

I wholeheartedly agree that we need to get into the test writing business. :+1:

File: .bandit.yml (hunk 1 above)
Comment (felddy) [reply .bandit.yml:14]: I don't like disabling it either, and I didn't know about the excludes at the pre-commit level.  This is the solution!   Will modify.  Thanks.

**Review by jsf9k — APPROVED** [review]
Great solution!

**Comment by felddy** [comment]
Check out this new thang:
image

**Comment by jsf9k** [comment]
> Check out this new thang:
> image

Yes, I saw that thang!

**Pull request merged**